### PR TITLE
Proper base classes for PorousFlow porosity, permeability

### PR DESCRIPTION
**Pull request by cpgr**
Provides base classes for porosity and permeability material in the `porous_flow` module, removing the double up of unused `Unity` versions and `Const` versions that can also provide material properties = 1. Previously there were some materials with base classes, some inheriting from `Unity` materials etc - a bit of a mess.

Also move the `PorousFlowRelativePermeabilityUnity` class to a `PorousFlowRelativePermeabilityConst` for consistency with other material properties.

Closes #7962

### CODE REVIEW

File: modules/porous_flow/src/materials/PorousFlowPorosityHM.C [hunk 1]
```diff
@@ -40,7 +38,7 @@ PorousFlowPorosityHM::PorousFlowPorosityHM(const InputParameters & parameters) :
     _pf_qp(getMaterialProperty<Real>("PorousFlow_effective_fluid_pressure_qp")),
     _dpf_qp_dvar(getMaterialProperty<std::vector<Real> >("dPorousFlow_effective_fluid_pressure_qp_dvar"))
 {
-  for (unsigned i = 0 ; i < _ndisp ; ++i)
+  for (unsigned int i = 0 ; i < _ndisp ; ++i)
```
Comment (andrsd) [inline modules/porous_flow/src/materials/PorousFlowPorosityHM.C]: No space in front of `;`

File: modules/porous_flow/src/materials/PorousFlowPorosityHM.C [hunk 2]
```diff
@@ -65,17 +63,17 @@ PorousFlowPorosityHM::computeQpProperties()
...
 
   _dporosity_qp_dgradvar[_qp].resize(_num_var);
   _dporosity_nodal_dgradvar[_qp].resize(_num_var);
-  for (unsigned v = 0; v < _num_var; ++v)
+  for (unsigned int v = 0; v < _num_var; ++v)
   {
-    _dporosity_qp_dgradvar[_qp][v] = -(_porosity_qp[_qp] - _biot) * _dvol_strain_qp_dvar[_qp][v];
-    _dporosity_nodal_dgradvar[_qp][v] = -(_porosity_nodal[_qp] - _biot) * _dvol_strain_qp_dvar[_qp][v];
+    _dporosity_qp_dgradvar[_qp][v] = - (_porosity_qp[_qp] - _biot) * _dvol_strain_qp_dvar[_qp][v];
+    _dporosity_nodal_dgradvar[_qp][v] = - (_porosity_nodal[_qp] - _biot) * _dvol_strain_qp_dvar[_qp][v];
```
Comment (andrsd) [inline modules/porous_flow/src/materials/PorousFlowPorosityHM.C]: We don't pad unary operators with spaces.

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/porous_flow/src/materials/PorousFlowPorosityHM.C (hunk 1 above)
Comment (cpgr) [inline modules/porous_flow/src/materials/PorousFlowPorosityHM.C]: Fixed

File: modules/porous_flow/src/materials/PorousFlowPorosityHM.C (hunk 2 above)
Comment (cpgr) [inline modules/porous_flow/src/materials/PorousFlowPorosityHM.C]: Oops - mustn't have been concentrating when I put those in. Fixed now.

**Review by andrsd — APPROVED** [review]

**Pull request merged**